src/metldata/builtin_transformations/common/assumptions.py

```diff
@@ -107,14 +110,28 @@ class in the provided `path_element`, depending on the type of the relation path
         )
 
 
+def assert_object_path_required(json_schema: Mapping[str, Any], path: str) -> None:
+    """Ensures that a path exists and its components are marked as 'required' in the schemapack.
+    This validates that any transformation relying on that path can depend on its presence
+    in a datapack.
+    """
+    if not path:
+        return
```

Comment:
Implementation here conflicts with the docstring.
Should this raise instead?

Reply:
No no, a path can be an empty string. If it is an empty string, the new property is directly added to the properties section of the content schema. We only need to check the assumption when a transformation relies on a certain path. I changed the doc string.